### PR TITLE
Db/update db models post sd1

### LINKED ISSUE #22: Update Database Entity Schemas
Update 12/16/19:

Updating User and Tank Models. Adding Marketplace Sale, Tank Component, Casus Block Models. Details based on SD1 Design document.

### PR DESCRIPTION
**Pull request by Derplime**
**Description**
Updating Database Schemas to reflect what was discussed in our big meeting and in our design document.

There is also an update to package-lock.json due to the recent additions of new/updated packages to help resolve flow/babel support. (i.e. I had `npm install`ed after bringing down the new package.json from `master` branch). If this update isn't needed, I can revert this since I thought ahead and made this change its own commit.

**Note:** I'd like everyone's input on these changes, especially from the those doing front-end/casus.
It's mostly to make sure that the models are structured to how you're expecting to use them. I'm not expecting this to go straight through. If you need specific things of certain models, please do suggest them.

Fixes #22 

**Type of change**
Check options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Fix or feature that would cause existing functionality to not work as expected (may need to update any API PRs if this goes in before the API)
- [x] This change requires a update in the design document (if changes outside these are asked for)

**Checklist:**
Go down the list and check them off.

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the design document (if applicable)
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by blaiadam** [comment]
Looks good to me! I'll wait for everyone to review before approving.

File: fortuna/package-lock.json [hunk 1]
```diff
@@ -5,20 +5,28 @@
   "requires": true,
   "dependencies": {
     "@babel/cli": {
-      "version": "7.7.0",
```
Comment (SecondThread) [inline fortuna/package-lock.json]: This stuff should be added to our .gitignore so it doesn't show up here

**Review by SecondThread — COMMENTED** [review]
Everything looks good to me except for the package-lock.json file which should be gitignore'd

File: fortuna/package-lock.json (hunk 1 above)
Comment (Derplime) [reply fortuna/package-lock.json]: I'll just revert the commit making the update to the package-lock. **However** I do have an issue with us ignoring it all-together. On that note, I'll bring that discussion to the Discord.

**Review by SecondThread — APPROVED** [review]
Thanks for fixing :)

**Review by JorgeVidal29 — APPROVED** [review]

**Review by Airor4 — APPROVED** [review]
Looks great. No questions as of now!

**Pull request merged**